  
  Use Position Independent Code for i586 also
  Otherwise, you will run into the following problem when linking
  (only lld warned about the issue, ld and ld.gold produced binaries
  that crashed in the dynamic linker during startup):
    ld: error: can't create dynamic relocation R_386_32 against local
      symbol in read only segment; recompile object files with -fPIC
      or pass '-Wl,-z,notext' to allow text relocations in the output
  
--- ./build/linux/unbundle/yasm.gn
+++ ./build/linux/unbundle/yasm.gn
@@ -4,6 +4,7 @@
 
 if (current_cpu == "x86") {
   _yasm_flags = [
+    "-DPIC",
     "-felf32",
     "-m",
     "x86",
--- ./third_party/openh264/BUILD.gn
+++ ./third_party/openh264/BUILD.gn
@@ -66,6 +66,7 @@
     defines = yasm_defines
     if (current_cpu == "x86") {
       defines += [ "X86_32" ]
+      defines += [ "X86_32_PICASM" ]
     } else {  # x64
       if (is_mac) {
         defines += [
@@ -87,6 +88,7 @@
     defines = yasm_defines
     if (current_cpu == "x86") {
       defines += [ "X86_32" ]
+      defines += [ "X86_32_PICASM" ]
     } else {  # x64
       if (is_mac) {
         defines += [
@@ -108,6 +110,7 @@
     defines = yasm_defines
     if (current_cpu == "x86") {
       defines += [ "X86_32" ]
+      defines += [ "X86_32_PICASM" ]
     } else {  # x64
       if (is_mac) {
         defines += [
@@ -136,6 +139,9 @@
   deps = []
   if (use_assembler) {
     defines = [ "X86_ASM" ]
+    if (current_cpu == "x86") {
+      defines += [ "X86_32_ASM" ]
+    }
     deps += [ ":openh264_common_yasm" ]
   }
   if (is_android) {
@@ -162,6 +168,9 @@
   deps = [ ":common" ]
   if (use_assembler) {
     defines = [ "X86_ASM" ]
+    if (current_cpu == "x86") {
+      defines += [ "X86_32_ASM" ]
+    }
     deps += [ ":openh264_processing_yasm" ]
   }
 }
@@ -188,6 +197,9 @@
   ]
   if (use_assembler) {
     defines = [ "X86_ASM" ]
+    if (current_cpu == "x86") {
+      defines += [ "X86_32_ASM" ]
+    }
     deps += [ ":openh264_encoder_yasm" ]
   }
 }
